google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/project/ActiveCloudProjectHolder.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.project;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import com.google.common.primitives.Longs;
+import com.intellij.ide.util.PropertiesComponent;
+import com.intellij.openapi.project.Project;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Holds active cloud project, one per IDE project. Active cloud project is the last cloud project
+ * selected by a user in most of the GCP related UI reported by {@link ProjectSelector}.
+ *
+ * <p>Must only be used on Swing/IDE EDT.
+ */
+public class ActiveCloudProjectHolder {
+  private static final String PROJECT_ACCOUNT_KEY = "ACTIVE_CLOUD_PROJECT_ACCOUNT";
+  private static final String PROJECT_NAME_KEY = "ACTIVE_CLOUD_PROJECT_NAME";
+  private static final String PROJECT_ID_KEY = "ACTIVE_CLOUD_PROJECT_ID";
+  private static final String PROJECT_NUMBER_KEY = "ACTIVE_CLOUD_PROJECT_NUMBER";
+
+  private static ActiveCloudProjectHolder instance = new ActiveCloudProjectHolder();
+
+  static ActiveCloudProjectHolder getInstance() {
+    return instance;
+  }
+
+  @VisibleForTesting
+  static void setInstance(ActiveCloudProjectHolder instance) {
+    ActiveCloudProjectHolder.instance = instance;
+  }
+
+  void setActiveCloudProject(
+      @NotNull CloudProject activeCloudProject, @NotNull Project ideProject) {
+    PropertiesComponent propertiesComponent = PropertiesComponent.getInstance(ideProject);
+
+    propertiesComponent.setValue(PROJECT_ACCOUNT_KEY, activeCloudProject.googleUsername());
+    propertiesComponent.setValue(PROJECT_NAME_KEY, activeCloudProject.projectName());
+    propertiesComponent.setValue(PROJECT_ID_KEY, activeCloudProject.projectId());
+    if (activeCloudProject.projectNumber() != null) {
```

Comment:
sorry if I've asked this before. but just so I understand: project number is nullable but the other values are not?

Reply:
Mostly for existing run/deploy/debug configurations compatibility. Only one that sets project number is debug configuration. Existing run/deploy configurations will have that as null until we expand them to contain complete project information (name, number, id, account).